server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -4490,7 +4493,7 @@ private Pair<List<TemplateJoinVO>, Integer> searchForTemplatesInternal(Long temp
             boolean showDomr, boolean onlyReady, List<Account> permittedAccounts, Account caller,
             ListProjectResourcesCriteria listProjectResourcesCriteria, Map<String, String> tags,
             boolean showRemovedTmpl, List<Long> ids, Long parentTemplateId, Boolean showUnique, String templateType,
-            Boolean isVnf) {
+            Boolean isVnf, Long domainId, boolean isRecursive) {
```

Comment:
I am not a fan of extending parameter lists beyond three. It already was, but we should think about passing the command or an interface the command adheres to instead. Not necessarily in this PR but keep in mind this is a maintainability issue. a lot of these are directly retrieved from the cmd object.